lib/swig/dynamic/golang.cgo.i

```diff
@@ -0,0 +1,96 @@
+/*GoSlice in typemap*/
+%typemap(in) GoSlice {
+	char* buffer = 0;
+	size_t size = 0;
+	int res = SWIG_AsCharPtrAndSize( $input, &buffer, &size, 0 );
```

Comment:
This whole file , even if it includes `SWIG_AsCharPtrAndSize`, should be implemented in `lib/swig/common` . Should a language be missing implementation of string typemap fragments then they should be added in client library repository .